modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinClientCodegen.java

```diff
@@ -149,6 +149,7 @@ public void processOpts() {
         if (CollectionType.LIST.value.equals(collectionType)) {
             typeMapping.put("array", "kotlin.collections.List");
             typeMapping.put("list", "kotlin.collections.List");
+            additionalProperties.put("isList", true);
```

Comment:
This `isList` seems redundant and potentially confusing with the already available `isListContainer`.

Reply:
`isListContainer` indicates that data is list or array but doesn't tell which type it should be. To propagate users preferences for `collectionType` to templates I've added this flag. Unfortunately mustashe templates doesn't have any mechanism for comparation, so it is boolean flag.

Reply:
Oh, I get it. You're providing a flag that says the generator is forcing arrays and lists to be of type `kotlin.collections.List`, where some properties and vars may be type array but not have `isListContainer=true`.  We could probably fix that in a model post process to make the model consistent, but this approach works as well.